x-pack/agent/docs/agent_configuration_example.yml

```diff
@@ -0,0 +1,209 @@
+    # Package this config group is coming from. On importing, we know where it belongs
+    # The package tells the UI which application to link to
+    package:
+      name: epm/nginx
+      version: 1.7.0
+    namespace?: prod
+    constraints:
+      # Contraints look are not final
+      - os.platform: { in: "windows" }
+      - agent.version: { ">=": "8.0.0" }
+    inputs:
+      - type: logs
+        processors?:
+        streams:
+          - id?: {id}
+            dataset: nginx.acccess
+            paths: /var/log/nginx/access.log
+          - id?: {id}
+            dataset: nginx.error
+            paths: /var/log/nginx/error.log
+      - type: metrics/nginx
+        streams:
+          - id?: {id}
+            dataset: nginx.stub_status
+            metricset: stub_status
+
+  #################################################################################################
+  # Custom Kafka datasource
+  - id: kafka-x1
+    title: "Consume data from kafka"
+    namespace?: prod
+    use_output: long_term_storage
+    inputs:
+      - type: kafka
+        host: localhost:6566
+        streams:
+          - dataset: foo.dataset
+            topic: foo
+            processors:
+             - extract_bro_specifics
+
+
+  #################################################################################################
+  # System EPM package
+  - id: system
+    title: Collect system information and metrics
+    package:
+      name: epm/system
+      version: 1.7.0
+    inputs:
+      - type: metrics/system
+        streams:
+         - id?: {id}
+           enabled?: false # default true
+           metricset: cpu
+           period: 10s
+           dataset: system.cpu
+           metrics: ["percentages", "normalized_percentages"]
+         - metricset: memory
+           dataset: system.memory
+         - metricset: diskio
+           dataset: system.diskio
+         - metricset: load
+           dataset: system.load
+         - metricset: memory
+           dataset: system.memory
+         - metricset: process
+           dataset: system.process
+           processes: ["firefox*"]
+         - metricset: process_summary
+           dataset: system.process_summary
+         - metricset: uptime
+           dataset: system.uptime
+         - metricset: socket_summary
+           dataset: system.socket_summary
+         - metricset: filesystem
+           dataset: system.filesystem
+         - metricset: raid
+           dataset: system.raid
+         - metricset: socket
+           dataset: system.socket
+         - metricset: service
+           dataset: system.service
+         - metricset: fsstat
+           dataset: system.fsstat
+         - metricset: foo
+           dataset: system.foo
+
+#################################################################################################
+### suggestion 1
+ - id: myendpoint-x1
+   title: Endpoint configuration
+   package:
+     name: endpoint
+     version: xxx
```

Comment:
We've discussed adding versions into the configuration that need to match up with Endpoint, otherwise, it will reject the configuration.  Is that what this `version` implies?  Will we be able to manage individual versions for each package? (i.e. endpoint, metric beat, etc)?

A similar conversation on versions here:
https://github.com/elastic/endpoint-app-team/issues/129#issuecomment-579130066

Reply:
This version here is to know from which package and package version a config was created. This should help when we try to upgrade configs. 

The part about limiting it to a specific agent version can be found here: https://github.com/elastic/beats/pull/15940/files#diff-043e80bbcc20fded11325ef31433396bR35